src/pybind/mgr/devicehealth/module.py

```diff
@@ -165,7 +165,7 @@ def is_valid_daemon_name(self, who: str) -> bool:
             return False
         return parts[0] in ('osd', 'mon')
 
-    @CLICommand('device query-daemon-health-metrics',
+    @CLIReadCommand('device query-daemon-health-metrics',
                 perm='r')
```

Comment:
If it a read only command then you replace like this
```suggestion
    @CLIReadCommand('device query-daemon-health-metrics')
```

Reply:
Understood!

Reply:
Understood!

Reply:
Understood!